ui/api/meshmodel.js

```diff
@@ -55,6 +55,12 @@ export async function getVersionedComponentFromModel(
   );
 }
 
+export async function getComponentsDetailWithPageSize(page=1, pageSize="all") {
+  return await promisifiedDataFetch(
+    `api/meshmodels/components?page=${page}&pagesize=${pageSize}`
+  );
+}
```

Comment:
I think we don't need to create separate function, we can use getComponentDetails by passing pageSize param with default value.

Reply:
Yes, that will be great. But since there were separate functions already present for **Mesh Models**, I followed the same convention.

https://github.com/meshery/meshery/blob/master/ui/api/meshmodel.js#L37

https://github.com/meshery/meshery/blob/master/ui/api/meshmodel.js#L64

Reply:
I think it is because the default page size is mentioned in **backend**. Let's say we modify the function with the **default page size**, then if the default page size is updated in the **backend**, the changes won't be reflected in the **front-end** unless we change it in front-end also.

Reply:
I think then we are good, @Chadha93 thoughts? ^^

Reply:
@duckling69?